gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -1869,20 +1900,62 @@ parse_accl_hw_all (const gchar * accelerators,
   return match_accl;
 }
 
+/**
+ * @brief Added basic accelerators (auto, default) to supported accelerators
+ * @note returned array must be freed by the caller
+ */
+static const gchar **
+add_basic_supported_accelerators (const gchar ** supported_accelerators)
+{
+  gint num_hw = 0, idx = 0;
+  const gchar **accl_support;
+
+  /** Count number of elements for the array */
+  while (supported_accelerators[num_hw] != NULL)
+    num_hw += 1;
+  num_hw += 2;
+
+  /** Allocate the array */
+  accl_support = g_malloc (sizeof (gchar *) * (num_hw + 1));
+
+  /** Fill the array */
+  while (supported_accelerators[idx] != NULL) {
+    accl_support[idx] = supported_accelerators[idx];
+    idx += 1;
+  }
+  accl_support[idx++] = ACCL_AUTO_STR;
+  accl_support[idx++] = ACCL_DEFAULT_STR;
+  accl_support[idx] = NULL;
+
+  return accl_support;
+}
+
 /**
  * @brief parse user given string to extract accelerator based on given regex
  * @param[in] accelerators user given input
  * @param[in] supported_accelerators list ofi supported accelerators
+ * @param[in] auto_accelerator accelerator to use in auto case (when acceleration is enabled but specific accelerator is not provided or not matching)
```

Comment:
It seems that user might unexpectedly encounter a situation that different accelerator is used against user's choice.
Is there any code bit that warn user when this happens?

Reply:
Good point, added at https://github.com/nnstreamer/nnstreamer/pull/2372/files#diff-97faf0d14d439e63e1544199a92ec25aR1891

Reply:
Thanks :)